install/kubernetes/values.yaml

```diff
@@ -158,19 +161,64 @@ tetragon:
   # host, the path is /proc. Exceptions to this are environments like kind, where the runtime itself
   # does not run on the host.
   hostProcPath: "/proc"
+
+# -- Tetragon Operator settings
 tetragonOperator:
+  # -- Annotations for the Tetragon Operator Deployment.
+  annotations: {}
+  # -- Annotations for the Tetragon Operator Deployment Pods.
+  podAnnotations: {}
+  # -- Extra labels to be added on the Tetragon Operator Deployment.
+  extraLabels: {}
+  # -- Extra labels to be added on the Tetragon Operator Deployment Pods.
+  extraPodLabels: {}
```

Comment:
The agent labels are configured with:
```
podLabels: {}
daemonSetLabelsOverride: {}
selectorLabelsOverride: {}
podLabelsOverride: {}
serviceLabelsOverride: {}
```
([L20-L24](https://github.com/cilium/tetragon/pull/1817/files#diff-be0f5319dfa71d552848e99593254e89869fdf280b90695ef6fb645544212538L20-L24)). `podLabels` is adding extra labels to pods and it seems there is no option to add extra labels to the daemonset, but there is `daemonSetLabelsOverride` to override all labels.

It bothers me a bit that these values are so inconsistent :( But I like yours more, they're more intuitive and I think more consistent with the whole ecosystem. So IMO it's fine to leave it like that, just wanted to point it out in case you think it's something worth consideration.

Reply:
Good point. I also saw those labels on the DS but used a different approach for the operator as it wasn't intuitive IMO. For example,
1. What's the reason why somebody would want to override selector labels within the DS (`selectorLabelsOverride`)?
2. One can configure `podLabels` and `podLabelsOverride` simultaneously, and the provided values will all be added. I mean, where's the override here? 😅
3. There's no way to add additional labels to the DS. You can either specify `daemonSetLabelsOverride` which leads to the situation that all (quite important) labels like `helm.sh/chart` and `app.kubernetes.io/*` won't be added or you can have only have those (`helm.sh/chart` and `app.kubernetes.io/*`).

I've got the feeling it would make sense to refactor the DS labels but most likely via separate PR and only with the next minor release as I'd consider this a (small) breaking change.